wpilibj/src/main/java/edu/wpi/first/wpilibj/smartdashboard/SendableChooser.java

```diff
@@ -171,4 +171,47 @@ public void initSendable(SendableBuilder builder) {
           }
         });
   }
+
+  /**
+   * Constructs a SendableChooser for an enum type. The returned chooser contains an option for each
+   * value of the enum.
+   *
+   * <p>Example:
+   *
+   * <pre>
+   * {@code SendableChooser<MyEnum> chooser = SendableChooser.forEnum(MyEnum.class); }
+   * </pre>
+   *
+   * @param <V> The enum type
+   * @param cls The class of the enum type
+   * @return A sendable chooser containing an options for each enum value.
+   */
+  public static <V extends Enum<V>> SendableChooser<V> forEnum(Class<V> cls) {
```

Comment:
From a code organization standpoint, these factory methods would be better defined after the constructors